network-mux/src/Network/Mux/Bearer/Socket.hs

```diff
@@ -44,34 +44,59 @@ import qualified Network.Mux.Time as Mx
 -- 'MuxError'.
 --
 socketAsMuxBearer
-  :: Tracer IO Mx.MuxTrace
+  :: Maybe DiffTime
+  -> Tracer IO Mx.MuxTrace
   -> Socket.Socket
   -> MuxBearer IO
-socketAsMuxBearer tracer sd =
+socketAsMuxBearer sduTimeout_m tracer sd =
       Mx.MuxBearer {
         Mx.read    = readSocket,
         Mx.write   = writeSocket,
         Mx.sduSize = 12288
       }
     where
+      hdrLenght = 8
+
       readSocket :: HasCallStack => IO (Mx.MuxSDU, Time)
       readSocket = do
           traceWith tracer $ Mx.MuxTraceRecvHeaderStart
-          hbuf <- recvLen' True 8 []
+
+          -- Wait for the first part of the header without any timeout
```

Comment:
Could you explain why we don't place a timeout on the `MuxSDU` header?

Reply:
Without it it would be impossible to implement timeouts that are larger at the miniprotocol layer than at the mux level. If we're waiting on a new header the peer isn't consuming any buffer space at the mux layer. As soon as we have received part of a mux SDU the peer is consuming resources and it should be limited.